src/lib/free-queue/README.md

```diff
@@ -1 +1,114 @@
-### Work In Progress ....
\ No newline at end of file
+# Audio Buffering Library
+
+## Introduction
+
+This library provides two classes, `FreeQueue` and `FreeQueueSAB`, designed to handle audio buffering using WebAssembly (WASM) and SharedArrayBuffer (SAB) respectively. These classes offer efficient ways to manage audio data, making them suitable for applications requiring low-latency audio processing.
+
+### Features
+
+- Efficient audio buffering using WASM (`FreeQueue`) and SharedArrayBuffer (`FreeQueueSAB`).
+- Supports multiple audio channels.
+- Designed for single-producer/single-consumer scenarios.
+- Provides atomic operations for safe access across threads.
+
+## Installation
+
+To use this library, you need to include it in your project. You can do this by cloning the repository and importing the classes into your project.
+
+```bash
+git clone <repository_url>
```

Comment:
Can this `repository_url` be just the URL of this project? Perhaps we can put it there directly?

Reply:
Yeah Sure!